source/s3_client.c

```diff
@@ -1476,8 +1481,19 @@ static void s_s3_client_on_acquire_http_connection(
             *current_http_connection = NULL;
         }
 
+        AWS_ASSERT(s_s3_max_request_count_per_connection > s_max_request_jitter_range);
+
+        uint32_t jitter_value = 0;
```

Comment:
until we implement a faster random number generator that isn't hitting an entropy source directly everytime, can you make this as small as possible? In this case a uint8_t would be plenty.